lib/inspec/resources/lxc.rb

```diff
@@ -0,0 +1,63 @@
+require "inspec/resources/command"
+module Inspec::Resources
+  class Lxc < Inspec.resource(1)
+    name "lxc"
+    # Restrict to only run on the below platforms
+    supports platform: "linux"
+    desc "Use the lxc InSpec audit resource to test if container exists and/or is running for linux container"
+    example <<~EXAMPLE
+      describe lxc("ubuntu-container") do
+        it { should exist }
+        it { should be_running }
+      end
+    EXAMPLE
+
+    # Resource initialization.
+    def initialize(container_name)
+      @container_name = container_name
+      return if  inspec.os.linux?
+
+      @container_info = []
+      skip_resource "The `lxc` resource is not supported on your OS yet."
```

Comment:
Seems like this `skip_resource` is called here, but it is never really checked if the resource is skipped to raise the exception? Have you tried this scenario? I feel this might not skip at all.

Reply:
Made the necessary changes [in this commit](https://github.com/inspec/inspec/pull/5921/commits/cee038c1f480dd275fded2fa2d2adee8ab69a400)